src/models/community_business.ts

```diff
@@ -272,5 +278,104 @@ export const CommunityBusinesses: CommunityBusinessCollection & CustomMethods =
 
     return query;
   },
+
+  async getVisitActivities (client: Knex, o: CommunityBusiness, d?: Day) {
+    const baseQuery = client('visit_activity')
+    .innerJoin(
+      'visit_activity_category',
+      'visit_activity_category.visit_activity_category_id',
+      'visit_activity.visit_activity_id')
+    .select({
+      id: 'visit_activity_id',
+      name: 'visit_activity_name',
+      category: 'visit_activity_category.visit_activity_category_name',
+      monday: 'monday',
+      tuesday: 'tuesday',
+      wednesday: 'wednesday',
+      thursday: 'thursday',
+      friday: 'friday',
+      saturday: 'saturday',
+      sunday: 'sunday',
+      createdAt: 'created_at',
+      modified_at: 'modified_at',
+    })
+      .where({ organisation_id: o.id, deleted_at: null });
+
+    const query = d
+      ? baseQuery.where({ [d]: true })
+      : baseQuery;
+
+    return query;
+  },
+
+  async addVisitActivity (client: Knex, v: Partial<VisitActivity>, c: Partial<CommunityBusiness>) {
+    const [res] = await client('visit_activity')
+    .insert({
+      visit_activity_name: v.name,
+      visit_activity_category_id: client('visit_activity_category')
+        .select('visit_activity_category_id')
+        .where({ visit_activity_category_name: v.category }),
+      organisation_id: c.id,
+    })
+    .returning([
+      'visit_activity_id AS id',
+      'visit_activity_name AS name',
+      'monday',
+      'tuesday',
+      'wednesday',
+      'thursday',
+      'friday',
+      'saturday',
+      'sunday',
+      'created_at AS createdAt',
+      'modified_at as modifiedAt',
+    ]);
+    return res || null;
```

Comment:
If the query returns an empty array, doesn't that mean the insert has failed? Are you sure you want to just return `null` in that case? Also this is inconsistent with the return type in the function signature `Promise<VisitActivity>`. This function has the return type `Promise<VisitActivity | null>`